flower_classifier/datasets/oxford_flowers.py

```diff
@@ -0,0 +1,93 @@
+from scipy.io import loadmat
+
+logger = logging.getLogger(__name__)
+
+
+NAMES = [
+    "pink primrose", "hard-leaved pocket orchid", "canterbury bells",
+    "sweet pea", "english marigold", "tiger lily", "moon orchid",
+    "bird of paradise", "monkshood", "globe thistle", "snapdragon",
+    "colt's foot", "king protea", "spear thistle", "yellow iris",
+    "globe-flower", "purple coneflower", "peruvian lily", "balloon flower",
+    "giant white arum lily", "fire lily", "pincushion flower", "fritillary",
+    "red ginger", "grape hyacinth", "corn poppy", "prince of wales feathers",
+    "stemless gentian", "artichoke", "sweet william", "carnation",
+    "garden phlox", "love in the mist", "mexican aster", "alpine sea holly",
+    "ruby-lipped cattleya", "cape flower", "great masterwort", "siam tulip",
+    "lenten rose", "barbeton daisy", "daffodil", "sword lily", "poinsettia",
+    "bolero deep blue", "wallflower", "marigold", "buttercup", "oxeye daisy",
+    "common dandelion", "petunia", "wild pansy", "primula", "sunflower",
+    "pelargonium", "bishop of llandaff", "gaura", "geranium", "orange dahlia",
+    "pink-yellow dahlia?", "cautleya spicata", "japanese anemone",
+    "black-eyed susan", "silverbush", "californian poppy", "osteospermum",
+    "spring crocus", "bearded iris", "windflower", "tree poppy", "gazania",
+    "azalea", "water lily", "rose", "thorn apple", "morning glory",
+    "passion flower", "lotus", "toad lily", "anthurium", "frangipani",
+    "clematis", "hibiscus", "columbine", "desert-rose", "tree mallow",
+    "magnolia", "cyclamen", "watercress", "canna lily", "hippeastrum",
+    "bee balm", "ball moss", "foxglove", "bougainvillea", "camellia", "mallow",
+    "mexican petunia", "bromelia", "blanket flower", "trumpet creeper",
+    "blackberry lily"
+]  # from https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/image_classification/oxford_flowers102.py
+IMAGE_URL = 'https://www.robots.ox.ac.uk/~vgg/data/flowers/102/102flowers.tgz'
+LABELS_URL = 'https://www.robots.ox.ac.uk/~vgg/data/flowers/102/imagelabels.mat'
+
+
+def download_dataset(root_dir: str):
+    root_dir = Path(root_dir)
+
+    # check if files already exist
+    images_exist = (root_dir / '102flowers.tgz').exists()
+    labels_exist = (root_dir / 'imagelabels.mat').exists()
+
+    # download and unpack images
+    if not images_exist:
+        logger.info('downloading images...')
+        torchvision.datasets.utils.download_and_extract_archive(IMAGE_URL, root_dir)
+
+    # download labels
+    if not labels_exist:
+        logger.info('downloading labels...')
+        torchvision.datasets.utils.download_url(LABELS_URL, root_dir)
+
+
+class OxfordFlowers102Dataset(Dataset):
+    """
+    Oxford 102 Category Flower Dataset
+    https://www.robots.ox.ac.uk/~vgg/data/flowers/102/index.html
+
+    Note: images are not shuffled by default.
+    """
+    def __init__(self,
+                 root_dir: str = '/content/drive/My Drive/Flowers/',
```

Comment:
Should we use the ROOT_DATA_DIR constant from the `__init__` module here instead of hard-coding the path?